mailbox/api/src/main/java/org/apache/james/mailbox/model/SimpleMailboxAnnotation.java

```diff
@@ -0,0 +1,328 @@
+import org.apache.james.mailbox.model.MailboxACL.MailboxACLEntryKey;
+
+import com.google.common.base.Preconditions;
+
+public class SimpleMailboxAnnotation implements MailboxAnnotation {
+
+    private static final Pattern NAME_ANNOTATION_PATTERN = Pattern.compile("[^\\x1a-\\x7f]");
+
+    public static final MailboxAnnotation EMPTY;
+
+    private final Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> entries;
+
+    static {
+        EMPTY = new SimpleMailboxAnnotation();
+    }
+
+    /**
+     * Creates a new instance of SimpleMailboxAnnotation containing no entries.
+     * 
+     */
+    public SimpleMailboxAnnotation() {
+        this.entries = Collections.emptyMap();
+    }
+
+    /**
+     * Creates a new instance of SimpleMailboxAnnotation from the given array of
+     * entries.
+     * 
+     * @param entries
+     */
+    public SimpleMailboxAnnotation(Map.Entry<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue>[] entries) {
+        if (entries != null) {
+            Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> m = 
+                    new HashMap<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue>(entries.length + entries.length / 2 + 1);
+            for (Entry<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> en : entries) {
+                m.put(en.getKey(), en.getValue());
+            }
+            this.entries = Collections.unmodifiableMap(m);
+        } else {
+            this.entries = Collections.emptyMap();
+        }
+    }
+
+    /**
+     * Creates a new instance of SimpleMailboxAnnotation from the given {@link Map} of
+     * entries.
+     * 
+     * @param entries
+     */
+    public SimpleMailboxAnnotation(Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> entries) {
+        if (entries != null && entries.size() > 0) {
+            Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> m = 
+                    new HashMap<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue>(entries.size() + entries.size() / 2 + 1);
+            for (Entry<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> en : entries.entrySet()) {
+                m.put(en.getKey(), en.getValue());
+            }
+            this.entries = Collections.unmodifiableMap(m);
+        } else {
+            this.entries = Collections.emptyMap();
+        }
+    }
+
+    /**
+     * Default implementation of {@link MailboxACLEntryKey}.
+     */
+    public static class SimpleMailboxAnnotationEntryKey implements MailboxAnnotationEntryKey {
+        private final String name;
+
+        /**
+         * Creates a new instance of SimpleMailboxAnnotationEntryKey from the given
+         * serialized {@link String}.
+         * 
+         * @param serialized
+         */
+        public SimpleMailboxAnnotationEntryKey(String serialized) {
+
+            if (serialized == null) {
+                throw new IllegalStateException("Cannot parse null to a " + getClass().getName());
```

Comment:
IllegalStateException is about state, here you check the argument, not the state, there's IllegalArgumentException for that.